### PR TITLE
Implement parsing function section

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

File: lib/fizzy/parser.hpp [hunk 1]
```diff
@@ -26,6 +26,12 @@ struct parser
 {
 };
 
+template <>
+struct parser<uint32_t>
```
Comment (chfast) [inline lib/fizzy/parser.hpp]: Why is this needed?

Comment (gumb0) [reply lib/fizzy/parser.hpp]: parser of vector requires parser of an element
